### PR TITLE
Migrate Finalizer upgrade test to use the upgrade framework

### PR DESCRIPTION
**Pull request by skmatti**

### CODE REVIEW

File: cmd/e2e-test/upgrade/finalizer.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		AddPath("foo.com", "/", svcName, port80).
+		Build()
+	ingKey := common.NamespacedName(ing)
+	fr.crud = e2e.IngressCRUD{C: fr.framework.Clientset}
+	if _, err := fr.crud.Create(ing); err != nil {
+		fr.t.Fatalf("error creating Ingress %s: %v", ingKey, err)
+	}
+	fr.t.Logf("Ingress created (%s/%s)", fr.s.Namespace, ing.Name)
+
+	if ing, err = e2e.WaitForIngress(fr.s, ing, &e2e.WaitForIngressOptions{ExpectUnreachable: true}); err != nil {
```
Comment (rramkumar1) [inline cmd/e2e-test/upgrade/finalizer.go]: Well now it seems like e2e.WaitForIngress() + s.PutStatus() is being used in multiple places. Can we add a wrapper for these two calls in pkg/e2e and maybe call it UpgradeTestWaitForIngress?

File: cmd/e2e-test/upgrade_test.go [hunk 2]
```diff
@@ -31,43 +31,60 @@ func TestGenericUpgrade(t *testing.T) {
...
-				// While k8s master is upgrading, it will return a connection refused
-				// error for any k8s resource we try to hit. We loop until the
-				// master upgrade has finished.
-				if s.MasterUpgrading() {
-					if err := test.DuringUpgrade(); err != nil {
-						t.Fatalf("For upgrade test %v, step DuringUpgrade failed due to %v", desc, err)
-					}
-					continue
-				}
+func genericUpgrade(t *testing.T, test e2e.UpgradeTest) {
```
Comment (rramkumar1) [inline cmd/e2e-test/upgrade_test.go]: runUpgradeTest

File: cmd/e2e-test/upgrade_test.go [hunk 3]
```diff
@@ -31,43 +31,60 @@ func TestGenericUpgrade(t *testing.T) {
...
-			func() {
-				// always mark the test as stable in order to unblock other upgrade tests.
-				defer s.PutStatus(e2e.Stable)
-				if err := test.PreUpgrade(); err != nil {
-					t.Fatalf("For upgrade test %v, step PreUpgrade failed due to %v", desc, err)
-				}
-			}()
+	for _, test := range []e2e.UpgradeTest{
+		upgrade.NewFinalizerUpgradeTest(),
+	} {
```
Comment (rramkumar1) [inline cmd/e2e-test/upgrade_test.go]: Don't need a for loop for now since you only have one case.

File: pkg/e2e/helpers.go [hunk 4]
```diff
@@ -90,6 +90,16 @@ func IsRfc1918Addr(addr string) bool {
 	return false
 }
 
+// UpgradeTestWaitForIngress waits for ingress to stabilize and set sandbox status to stable.
```
Comment (rramkumar1) [inline pkg/e2e/helpers.go]: Add a note to the comment that states that this is only used in upgrade tests similar to what you have below.

**Comment by rramkumar1** [comment]
Looks good to me. Will officially approve once that last comment is addressed.

File: pkg/e2e/helpers.go (hunk 4 above)
Comment (skmatti) [reply pkg/e2e/helpers.go]: Done

**Pull request merged**